src/domain/orders/index.tsx

```diff
@@ -1,41 +1,95 @@
 import { RouteComponentProps, Router } from "@reach/router"
-import React, { useMemo } from "react"
 import { navigate } from "gatsby"
+import { useAdminCreateBatchJob } from "medusa-react"
+import React, { useMemo } from "react"
+import Button from "../../components/fundamentals/button"
+import ExportIcon from "../../components/fundamentals/icons/export-icon"
 import BodyCard from "../../components/organisms/body-card"
 import TableViewHeader from "../../components/organisms/custom-table-header"
+import ExportModal from "../../components/organisms/export-modal"
 import OrderTable from "../../components/templates/order-table"
+import useNotification from "../../hooks/use-notification"
+import useToggleState from "../../hooks/use-toggle-state"
+import { getErrorMessage } from "../../utils/error-messages"
 import Details from "./details"
 
 const VIEWS = ["orders", "drafts"]
 
 const OrderIndex: React.FC<RouteComponentProps> = () => {
   const view = "orders"
 
+  const createBatchJob = useAdminCreateBatchJob()
+  const notification = useNotification()
+
+  const {
+    open: openExportModal,
+    close: closeExportModal,
+    state: exportModalOpen,
+  } = useToggleState(false)
+
   const actions = useMemo(() => {
-    return []
+    return [
+      <Button
+        variant="secondary"
+        size="small"
+        onClick={() => openExportModal()}
+      >
+        <ExportIcon size={20} />
+        Export Orders
+      </Button>,
+    ]
   }, [view])
 
+  const handleCreateExport = () => {
+    const reqObj = {
+      type: "order-export",
+      context: {},
+      dry_run: false,
+    }
+
+    createBatchJob.mutate(reqObj, {
+      onSuccess: () => {
+        notification("Success", "Successfully initiated export", "success")
+      },
+      onError: (err) => {
+        notification("Error", getErrorMessage(err), "error")
+      },
+    })
+
+    closeExportModal()
```

Comment:
question: Since we are not awaiting for the creation, could it be possible that the modal is closed before the notification is displayed? is that intended? Maybe that we should call the closeExportModal on success?